### PR TITLE
Removed uninterfaced COS2 non-bonded interaction

### PR DESCRIPTION
**Pull request by fweik**

### CODE REVIEW

File: src/core/CMakeLists.txt [hunk 1]
```diff
@@ -100,8 +100,7 @@ set(EspressoCore_SRC
   immersed_boundary/ImmersedBoundaries.cpp
   nonbonded_interactions/bmhtf-nacl.cpp
   nonbonded_interactions/buckingham.cpp
-  nonbonded_interactions/cos2.cpp
-  nonbonded_interactions/gaussian.cpp
+        nonbonded_interactions/gaussian.cpp
```
Comment (pkreissl) [inline src/core/CMakeLists.txt]: Why this indent?

**Review by pkreissl — DISMISSED** [review]

**Comment by jngrad** [comment]
please also remove `COS2` at lines 541-542 of /doc/sphinx/installation.rst

File: src/core/CMakeLists.txt (hunk 1 above)
Comment (fweik) [reply src/core/CMakeLists.txt]: ide...

**Review by jngrad — APPROVED** [review]

**Comment by fweik** [comment]
bors r=jngrad

**Pull request merged**